compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ClassCodegen.kt

```diff
@@ -82,6 +84,11 @@ open class ClassCodegen protected constructor(
             else -> null
         }
 
+    fun getRegeneratedObjectNameGenerator(function: Name) =
+        context.regeneratedObjectNameGenerators.getOrPut(irClass to function) {
```

Comment:
Actually it's possible to keep name generator in ClassCodegen with class related prefix (it would be more local) and for other purposes create subgenerator via NameGenerator.subGenerator

Reply:
That's unfortunately what the new test (boxInline/lambdaTransformation/regeneratedLambdaName2.kt) is about. Since currently IrSourceCompilerForInline creates a new (Fake)ClassCodegen for every call to an inline function for which source code is available, caching name generators at ClassCodegen level would cause the second call to `sameName` to overwrite the anonymous object created by the first one with a different implementation.

Reply:
Added a TODO about this.